tests/system/python/packages/test_authentication.py

```diff
@@ -0,0 +1,1157 @@
+
+
+@pytest.fixture
+def change_to_auth_mandatory_certificate(foglamp_url, wait_time):
+    # Wait for foglamp server to start
+    time.sleep(wait_time)
+    conn = http.client.HTTPConnection(foglamp_url)
+    conn.request("PUT", '/foglamp/category/rest_api',
+                 json.dumps({"authentication": "mandatory", "authMethod": "certificate"}))
+    r = conn.getresponse()
+    assert 200 == r.status
+    r = r.read().decode()
+    jdoc = json.loads(r)
+    assert "mandatory" == jdoc['authentication']['value']
+
+    conn.request("PUT", '/foglamp/restart', json.dumps({}))
+    r = conn.getresponse()
+    assert 200 == r.status
+    r = r.read().decode()
+    jdoc = json.loads(r)
+    assert "FogLAMP restart has been scheduled." == jdoc['message']
+
+
+@pytest.fixture
+def reset_foglamp():
+    # TODO: Remove kill after resolution of FOGL-1499
+    try:
+        subprocess.run(["$FOGLAMP_ROOT/bin/foglamp kill"], shell=True, check=True)
+    except subprocess.CalledProcessError:
+        assert False, "reset package script failed!"
+
+    try:
+        subprocess.run(["cd {}/tests/system/python/scripts/package && ./reset"
+                       .format(PROJECT_ROOT)], shell=True, check=True)
+    except subprocess.CalledProcessError:
+        assert False, "reset package script failed!"
+
+
+@pytest.fixture
+def remove_add_foglamp_pkgs():
+    try:
+        subprocess.run(["cd {}/tests/system/python/scripts/package && ./remove"
+                       .format(PROJECT_ROOT)], shell=True, check=True)
+    except subprocess.CalledProcessError:
+        assert False, "remove package script failed!"
+
+    try:
+        subprocess.run(["cd {}/tests/system/python/scripts/package/ && ./setup {}"
+                       .format(PROJECT_ROOT, package_build_version)], shell=True, check=True)
+    except subprocess.CalledProcessError:
+        assert False, "setup package script failed"
+
+    try:
+        subprocess.run(["sudo apt install -y foglamp-south-http-south"], shell=True, check=True)
+    except subprocess.CalledProcessError:
+        assert False, "installation of packages failed"
+
+
+class TestTLSDiasbled:
+    def test_on_default_port(self, set_project_root, reset_foglamp, foglamp_url, wait_time):
+        conn = http.client.HTTPConnection(foglamp_url)
+        conn.request("GET", "/foglamp/ping")
+        r = conn.getresponse()
+        jdoc = json.loads(r.read().decode())
+        assert "uptime" in jdoc
+        assert 0 < jdoc['uptime'], "FogLAMP not up."
+
+    def test_on_custom_port(self, foglamp_url, wait_time):
+        conn = http.client.HTTPConnection(foglamp_url)
+        conn.request("PUT", '/foglamp/category/rest_api', json.dumps({"httpPort": "8005"}))
+        r = conn.getresponse()
+        assert 200 == r.status
+        r = r.read().decode()
+        jdoc = json.loads(r)
+        assert "httpPort" in jdoc
+        assert '8005' == jdoc['httpPort']['value']
+
+        time.sleep(wait_time)
```

Comment:
why you need to wait before restart? Always put a explanatory comment before  `time.sleep`